xclim/indices/_agro.py

```diff
@@ -341,3 +344,71 @@ def water_budget(
 
     out.attrs["units"] = pr.attrs["units"]
     return out
+
+
+@declare_units(pr="[precipitation]", thresh="[precipitation]")
```

Comment:
To make clear that the threshold is on the _accumulated_ precipitation, I believe it should be `thresh="[length]"`, there is no error because we are very permissive for "hydro" units!

You might want to use `xc.core.units.rate2amount` on `pr` for the same reason.

Reply:
+1 for this